packages/sst/src/constructs/AstroSite.ts

```diff
@@ -174,6 +177,56 @@ export class AstroSite extends SsrSite {
           origin: "staticsServer",
         });
       } else {
+        if (buildMeta.imageService === "sharp") {
+          const pathToSharpLayerZip = createRequire(import.meta.url).resolve(
+            "astro-sst/layers/sharp"
+          );
```

Comment:
Does it make sense to have `astro-sst` spit out sharp into `dist`, and `AstroSite` can find it at `join(sitePath, "dist", "image", "sharp")`.
```
my-astro-app/
  dist/
    server/
    client/
    image/
```

Reply:
I had considered this and was part of the way to implementing it that way but I decided it was doing a lot of work for no benefit. The layer is not exactly tiny (15.6 MB), so copying it over on every rebuild seems a bit excessive. Also, the phase when it would be copied over happens before the final CDK evaluation so the layer could be moved over, but then not used because a different server config or layer is used. It just seemed a bit cleaner to leave it in place.

All that being said, it does feel odd to be deploying a file directly from an `npm` package.

Reply:
Makes sense.

Taking a step back, currently when building `AstroSite`'s server function, SST already detects if `sharp` is being used. If detected, `sharp` gets automatically packaged into the function bundle.
https://github.com/sst/sst/blob/b79a68a3d1f01a6983337cd950ba1b7ae838a045/packages/sst/src/runtime/handlers/node.ts#L209-L219

ie. I just ran `sst build`, and the server bundle has a `node_module` folder w/ `sharp` in it
<img width="543" alt="Screen Shot 2023-11-06 at 7 26 58 PM" src="https://github.com/sst/sst/assets/83515/96a0192a-2653-412b-9f90-219d2f480db5">


I don't think we need the layer?

Reply:
The layer does still help for startup time (supposedly), but the biggest thing is that it prevents a lot of really common deployment issues because the runtime binaries for `sharp` need to be Linux and most developers are using OSX or Windows, so this is a common issue that results in devs uninstalling and reinstalling packages before each deployment, and reverting again afterwards.

I'd certainly like to avoid any duplication of code/resources on the instance, so it's worth looking into further. I've tested and confirmed that both the layer and the `nodejs.esbuild.external["sharp"]` declaration are necessary. Removing either or both results in 500 with an error about not finding `sharp`.

Reply:
@fwang, you are correct that the `sharp` dependencies were being built into the bundle, but they weren't working in many cases because the architecture of the binaries did not match the runtime.

I've fixed this part of this PR, so now both layers and runtime bundles are supported.